src/responses.rs

```diff
@@ -16,6 +18,26 @@ pub struct OutpackError {
     pub detail: String,
 }
 
+impl OutpackError {
+    pub fn new(e: io::Error) -> OutpackError {
```

Comment:
So we'd have to extend this whenever a known error of a different type could have occurred